File: src/main/java/cn/lnkdoc/sdk/uia/common/client/IUiaClient.java
package cn.lnkdoc.sdk.uia.common.client;

import cn.lnkdoc.sdk.uia.common.exception.UiaException;
import cn.lnkdoc.sdk.uia.common.request.IUiaRequest;
import cn.lnkdoc.sdk.uia.common.response.IUiaResponse;

/**
 * @author langkye
 * @since 1.0.0.RELEASE
 */
public interface IUiaClient {
    /**
     * execute
     *
     * @param <RESP> type
     * @param request request implementation
     * @return response implementation`
     * @throws UiaException UiaException
     */
    <RESP> IUiaResponse<RESP> execute(IUiaRequest request) throws UiaException;

}

File: awesome-uia-sdk-common/awesome-uia-sdk-core/src/main/kotlin/cn/lnkdoc/sdk/uia/common/client/IUiaClient.kt
package cn.lnkdoc.sdk.uia.common.client

import cn.lnkdoc.sdk.uia.common.exception.UiaException
import cn.lnkdoc.sdk.uia.common.request.IUiaRequest
import cn.lnkdoc.sdk.uia.common.response.IUiaResponse

/**
 * @author langkye
 * @since 1.0.0.RELEASE
 */
interface IUiaClient {
    /**
     * execute
     *
     * @param <RESP> type
     * @param request request implementation
     * @return response implementation`
     * @throws UiaException UiaException
     */
    @Throws(UiaException::class)
    fun <RESP> execute(request: IUiaRequest): IUiaResponse<RESP>
}